src/components/application_manager/src/commands/mobile/create_interaction_choice_set_request.cc

```diff
@@ -267,12 +344,33 @@ void CreateInteractionChoiceSetRequest::SendVRAddCommandRequests(
       }
     }
 
-    msg_params[strings::cmd_id] =
-        choice_set[strings::choice_set][chs_num][strings::choice_id];
-    msg_params[strings::vr_commands] =
-        smart_objects::SmartObject(smart_objects::SmartType_Array);
-    msg_params[strings::vr_commands] =
-        choice_set[strings::choice_set][chs_num][strings::vr_commands];
+    const SmartObject& choice = choice_set[strings::choice_set][chs_num];
+
+    if (!choice[strings::vr_commands].asArray() ||
```

Comment:
@LevchenkoS 
`!choice[strings::vr_commands].asArray() ` looks quite strange.
Ok `asArray` returns pointer but is is not clear from this call.
I would suggest you to split ths logic